cli/src/commands/content/deleteVideoAssetsAsModerator.ts

```diff
@@ -0,0 +1,78 @@
+import { createType } from '@joystream/types'
+import { flags } from '@oclif/command'
+import { formatBalance } from '@polkadot/util'
+import chalk from 'chalk'
+import ContentDirectoryCommandBase from '../../base/ContentDirectoryCommandBase'
+import ExitCodes from '../../ExitCodes'
+import BN from 'bn.js'
+
+export default class DeleteVideoAssetsAsModeratorCommand extends ContentDirectoryCommandBase {
+  static description = 'Delete the video assets.'
+
+  protected requiresQueryNode = true
+
+  static flags = {
+    videoId: flags.integer({
+      char: 'v',
+      required: true,
+      description: 'ID of the Video',
+    }),
+    assetIds: flags.string({
+      char: 'a',
+      description: `List of data object IDs to delete`,
+      required: true,
+      multiple: true,
+    }),
+    rationale: flags.string({
+      char: 'r',
+      required: true,
+      description: 'Reason for removing the video assets by moderator',
+    }),
+    ...ContentDirectoryCommandBase.flags,
+  }
+
+  async getDataObjectsInfo(videoId: number, assetIds: string[]): Promise<[string, BN][]> {
+    const dataObjects = await this.getQNApi().dataObjectsByVideoId(videoId.toString())
+
+    return assetIds.map((id) => {
+      const dataObject = dataObjects.find((o) => o.id === id)
+      if (dataObject) {
+        return [dataObject.id, new BN(dataObject.stateBloatBond)]
+      }
+
+      this.error(`Data object ${id} is not associated with video ${videoId}`, {
+        exit: ExitCodes.InvalidInput,
+      })
+    })
+  }
+
+  async run(): Promise<void> {
+    const {
+      flags: { videoId, assetIds, rationale },
+    } = this.parse(DeleteVideoAssetsAsModeratorCommand)
+    // Context
+    const [actor, address] = await this.getCuratorContext()
```

Comment:
Also perhaps in case of curator you could add a check whether curator has the right moderation permissions? (based on group's `ModerationPermissionsByLevel` and channel's privilege level)